examples/bert/src/bert_layers.py

```diff
@@ -474,7 +481,18 @@ def forward(
         #        hidden_states[ntokens,hidden] -> hidden_states[ntokens_unpad,hidden]
         hidden_states, indices, cu_seqlens, _ = unpad_input(
             hidden_states, attention_mask_bool)
+
         # Add alibi matrix to extended_attention_mask
+        if self._current_alibi_size < seqlen or self.alibi is None:
+            # Build/rebuild the alibi tensor if needed
+            self.alibi = self.build_alibi_tensor(
+                n_heads=self.num_attention_heads,
+                max_token_length=seqlen,
+                device=hidden_states.device)
+            self._current_alibi_size = int(seqlen)
+        if self.alibi.device != hidden_states.device:
+            # Device catch-up
```

Comment:
For a noob like me, can you clarify that this makes sure that the alibi mask is synced with the correct device(s)? Why wouldn't it be otherwise?

Reply:
Yes, that is what's happening! We're assuming that the inputs to the model are on the correct device, so we're using `hidden_states.device` to get that device.

The alibi mask might not be on the correct device if something caused `self.alibi` to be created (i.e. the model's forward method was called) in between the model being on CPU and the model being moved to another device. In that case, `self.alibi` would still be on CPU and the device would need to "catch up" with the rest of the model :)